bazel/emscripten_deps.bzl

```diff
@@ -40,15 +59,33 @@ filegroup(
         "bin/wasm-emscripten-finalize{bin_extension}",
         "bin/wasm-ld{bin_extension}",
         "bin/wasm-opt{bin_extension}",
-   ] + glob(["emscripten/node_modules/**"]),
+        "bin/wasm-metadce{bin_extension}",
+        ":emcc",
+    ] + glob(
+        include = [
+            "emscripten/cache/sysroot/lib/**",
+            "emscripten/node_modules/**",
+            "emscripten/src/**",
+        ],
+    ),
 )
 
 filegroup(
     name = "ar_files",
     srcs = [
-        "emscripten/emar.py",
         "bin/llvm-ar{bin_extension}",
-   ],
+        "emscripten/emar.py",
+        "emscripten/emscripten-version.txt",
+        "emscripten/src/settings.js",
+        "emscripten/src/settings_internal.js",
+    ] + glob(
+        include = [
+            "emscripten/tools/**",
```

Comment:
If you want to you could probably exclude `tools/mains/`, `tools/experimental`, `tools/scons`, and `tools/websocket_to_posix_proxy/`, although they are not very big.